libraries/AP_Arming/AP_Arming.cpp

```diff
@@ -493,6 +504,81 @@ bool AP_Arming::manual_transmitter_checks(bool report)
     return true;
 }
 
+bool AP_Arming::mission_checks(bool report)
+{
+    if (((checks_to_perform & ARMING_CHECK_ALL) || (checks_to_perform & ARMING_CHECK_MISSION)) &&
+        _required_mission_items) {
+        AP_Mission *mission = AP::mission();
+        if (mission == nullptr) {
+            check_failed(ARMING_CHECK_MISSION, report, "No mission library present");
+            #if CONFIG_HAL_BOARD == HAL_BOARD_SITL
+                AP_HAL::panic("Mission checks requested, but no mission was allocated");
+            #endif // CONFIG_HAL_BOARD == HAL_BOARD_SITL
+            return false;
+        }
+        AP_Rally *rally = AP::rally();
+        if (rally == nullptr) {
+            check_failed(ARMING_CHECK_MISSION, report, "No rally library present");
+            #if CONFIG_HAL_BOARD == HAL_BOARD_SITL
+                AP_HAL::panic("Mission checks requested, but no rally was allocated");
+            #endif // CONFIG_HAL_BOARD == HAL_BOARD_SITL
+            return false;
+        }
+        for (int mask = 1; mask < MIS_ITEM_CHECK_MAX; mask <<= 1) {
```

Comment:
This stuff would be a lot shorter if we factored https://github.com/ardupilot/ardupilot/blob/master/libraries/AP_Mission/AP_Mission.cpp#L1815 and used it in a `default:` (or stacked `case`s)

Reply:
I'm not following, can you expand on that? As far as I can tell those are very different use cases.

Reply:
See the pair of commits here: https://github.com/peterbarker/ardupilot/commits/wickedshell/arming-mission (untested, very rough!)

Reply:
FWIW that's actually a larger build :D (252 bytes). With that being said I do see the appeal/makes it easy to add more individual checks in the future. Are we fine with the increased size vs clarity perspective here?